### PR TITLE
job-info: Add guest access to eventlogs

### PR DESCRIPTION
**Pull request by chu11**
Per discussion i #2041.  I know it's not much, but wanted to get the structure of this in as well as the tests.  The tests fake job submission under a guest account for the time being.

Nice chunk of code lifted from @garlick's branch in https://github.com/garlick/flux-core/tree/flux_job_eventlog

### CODE REVIEW

File: src/modules/job-info/job-info.c [hunk 1]
```diff
@@ -14,6 +14,8 @@
 #include <czmq.h>
 #include <jansson.h>
 #include <flux/core.h>
+#include <argz.h>
```
Comment (garlick) [inline src/modules/job-info/job-info.c]: argz/envz probably not needed anymore

**Comment by garlick** [comment]
Was just beginning to comment:
> Oh yeah, on the ci failures:  when running `--with-flux-security`, the ingest module checks that the signed jobspec matches the authenticated user.  Those env vars are influencing the authenticated user but the signature comes from munge which can't be spoofed.

but it looks like you just pushed changes to that effect?

**Comment by chu11** [comment]
> Oh yeah, on the ci failures: when running --with-flux-security, the ingest module checks that the signed jobspec matches the authenticated user. Those env vars are influencing the authenticated user but the signature comes from munge which can't be spoofed.  but it looks like you just pushed changes to that effect?

I'm not 100% following.  I was aware of the `job-ingest` module not working with the USERID & ROLEMASK environment variable test hacks.  So in the tests, job submission does not use the environment variables.  The environment variable hacks are only done on the `flux job eventlog` calls.

Most of the test failures appear to be a portability issue w/ `printf` and the `-v` option.

**Comment by chu11** [comment]
re-pushed hoping

```
+        eventlognew=$(echo -e "${eventlogsub}"'\nX')
+        eventlognew=${eventlognew%?}
```

is a portable way to add a newline to a string.

**Comment by chu11** [comment]
@garlick I just noticed the `flux-job: 0: signer=2000 != requestor=9000` error in one of the CI tests.  That's what I assume you were referring to.  What's interesting is it only occurs on one test, not both of the times I try to submit.  I'm guessing the first test fails and does not clear the environment variables.  Leading to the issue on the second test.

My fix newline fix above did not appear to work.  Ugh ...

**Comment by chu11** [comment]
ok, lets try

```
+        eventlognew=$(printf "${eventlogsub}"'\n'"X")
+        eventlognew=${eventlognew%?}
```

**Comment by garlick** [comment]
On my earlier comment re: ci failures:  I had gone back to look at code I had briefly scanned and thought it had changed out from under me.  Probably what happened was I misinterpreted it on first glance.  Sorry for the noise!

**Comment by chu11** [comment]
rebased

**Comment by chu11** [comment]
hmmm, got a valgrind error, but not much to go on.  I didn't get this error before I rebased.

```
expecting success: 
	run_timeout 120 \
	flux start -s ${VALGRIND_NBROKERS} --wrap=libtool,e,${VALGRIND} \
		--wrap=--tool=memcheck \
		--wrap=--leak-check=full \
		--wrap=--gen-suppressions=all \
		--wrap=--trace-children=no \
		--wrap=--child-silent-after-fork=yes \
		--wrap=--num-callers=30 \
		--wrap=--leak-resolution=med \
		--wrap=--error-exitcode=1 \
		--wrap=--suppressions=$VALGRIND_SUPPRESSIONS \
		-o,--shutdown-grace=${VALGRIND_SHUTDOWN_GRACE} \
		 ${VALGRIND_WORKLOAD}

flux-start: warning: setting --bootstrap=selfpmi due to --size option
==10922== Memcheck, a memory error detector
==10922== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==10922== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==10922== Command: /usr/src/src/broker/.libs/flux-broker --setattr=broker.rundir=/tmp/flux-10876-ePezWi/1 --setattr=tbon.endpoint=ipc://%B/req --shutdown-grace=16
==10922== 
==10921== Memcheck, a memory error detector
==10921== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==10921== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==10921== Command: /usr/src/src/broker/.libs/flux-broker --setattr=broker.rundir=/tmp/flux-10876-ePezWi/0 --setattr=tbon.endpoint=ipc://%B/req --shutdown-grace=16 /usr/src/t/valgrind/valgrind-workload.sh
==10921== 
FLUX_URI=local:///tmp/flux-10876-ePezWi/0
Running job
id=40248541184
id=46053457920
id=54911827968
id=61924704256
id=68182605824
id=75363254272
id=82409684992
id=89489670144
id=99069460480
id=105478356992
1553551433.554582 submit userid=2000 priority=16 flags=0
1553551438.502286 alloc note="rank0/core0"
1553551438.795803 start
1553551438.809766 finish status=0
1553551439.117221 release ranks="all" final=true
1553551439.147715 free
1553551439.148024 clean
++ flux job submit
++ flux jobspec srun -t 1 -n 1 /bin/true
+ id=208574349312
+ flux job wait-event 208574349312 start
1553551440.206890 start
+ flux job cancel 208574349312
+ flux job wait-event 208574349312 clean
1553551440.939137 clean
==10922== 
==10922== HEAP SUMMARY:
==10922==     in use at exit: 22,695 bytes in 58 blocks
==10922==   total heap usage: 32,636 allocs, 32,578 frees, 18,519,522 bytes allocated
==10922== 
==10922== LEAK SUMMARY:
==10922==    definitely lost: 0 bytes in 0 blocks
==10922==    indirectly lost: 0 bytes in 0 blocks
==10922==      possibly lost: 0 bytes in 0 blocks
==10922==    still reachable: 22,695 bytes in 58 blocks
==10922==         suppressed: 0 bytes in 0 blocks
==10922== Reachable blocks (those to which a pointer was found) are not shown.
==10922== To see them, rerun with: --leak-check=full --show-leak-kinds=all
==10922== 
==10922== For counts of detected and suppressed errors, rerun with: -v
==10922== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
flux-start: 0 (pid 10921) Killed
not ok 1 - valgrind reports no new errors on 2 broker run
```

**Comment by grondo** [comment]
```
flux-start: 0 (pid 10921) Killed
```

Looks like the killer-timeout exceeded?

File: src/modules/job-info/job-info.c [hunk 2]
```diff
@@ -128,6 +129,78 @@ static int lookup_key (struct lookup_ctx *l)
...
+    }
+    if (json_unpack (o, "{ s:i }", "userid", useridp) < 0) {
+        errno = EPROTO;
+        goto error;
+    }
+    free (event);
+    json_decref (o);
+    return 0;
+ error:
+    free (event);
```
Comment (garlick) [inline src/modules/job-info/job-info.c]: I think we need to save / restore errno in this block.

**Comment by garlick** [comment]
Apologies, just noticed a small nit that I missed before and commented above.

I think it would be good actually to get #2104 in before this one if you have a sec to go look at it.  @grondo and I both reviewed it, but we both contributed so it would be better form to have someone else  push the button.

File: src/modules/job-info/job-info.c (hunk 2 above)
Comment (chu11) [reply src/modules/job-info/job-info.c]: ahhh yes, when we converted to json formatted contexts the json_decref was added.  Thanks!

**Comment by chu11** [comment]
rebased

**Comment by chu11** [comment]
hmmm.  diff coverage isn't as good as I'd like it to be.  With the recent change fell below 70%.  Let me add some fake job submissions to try and eek out some extra percentage points.

**Comment by chu11** [comment]
re-pushed with some extra tests to hopefully cover more error paths.  Maybe it's excessive.

**Comment by chu11** [comment]
i already changed my mind, I added too many error path tests, cut it back to something more reasonable.  Just want to get close to 80% coverage.

**Comment by chu11** [comment]
woo, hoo, 88% coverage

**Comment by chu11** [comment]
rebased

**Comment by garlick** [comment]
Awesome!  Thanks @chu11!

**Pull request merged**